yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/launcher/DockerComputerJNLPLauncher.java

```diff
@@ -75,12 +86,55 @@ public JNLPLauncher getJnlpLauncher() {
     }
 
     @DataBoundSetter
-    public void setUser(String user) {
-        this.user = user;
+    public void setSlaveOpts(@Nonnull String slaveOpts) {
+        this.slaveOpts = StringUtils.trimToEmpty(slaveOpts);
+    }
+
+    public String getSlaveOpts() {
+        return StringUtils.trimToEmpty(slaveOpts);
+    }
+
+    @DataBoundSetter
+    public void setJenkinsUrl(@Nonnull String jenkinsUrl) {
+        this.jenkinsUrl = StringUtils.trimToEmpty(jenkinsUrl);
+    }
+
+    public String getJenkinsUrl() {
+        return StringUtils.trimToEmpty(jenkinsUrl);
+    }
+
+    public String getJenkinsUrl(@Nonnull String rootUrl) {
+        if (!isNull(jenkinsUrl) && StringUtils.isNotEmpty(jenkinsUrl)) {
```

Comment:
and then you can oneline to
` return isNotEmpty(jenkinsUrl) ? jenkinsUrl : trimToEmpty(rootUrl)`